src/core/client_channel/client_channel_filter.cc

```diff
@@ -2731,28 +2730,29 @@ class ClientChannelFilter::LoadBalancedCall::BackendMetricAccessor final
 
 namespace {
 
-void CreateCallAttemptTracer(grpc_call_context_element* context,
-                             bool is_transparent_retry) {
-  auto* call_tracer = static_cast<ClientCallTracer*>(
-      context[GRPC_CONTEXT_CALL_TRACER_ANNOTATION_INTERFACE].value);
+void CreateCallAttemptTracer(Arena* arena, bool is_transparent_retry) {
+  auto* call_tracer = DownCast<ClientCallTracer*>(
+      arena->GetContext<CallTracerAnnotationInterface>());
   if (call_tracer == nullptr) return;
   auto* tracer = call_tracer->StartNewAttempt(is_transparent_retry);
-  context[GRPC_CONTEXT_CALL_TRACER].value = tracer;
+  arena->SetContext<CallTracerInterface>(tracer);
 }
 
 }  // namespace
 
 ClientChannelFilter::LoadBalancedCall::LoadBalancedCall(
     ClientChannelFilter* chand, grpc_call_context_element* call_context,
```

Comment:
I assume there's some future PR where `grpc_call_context_element` goes away entirely, including removing this parameter?

Reply:
Yeah, it's all prepped... first we nuke the usages though, then the bulk cleanup (incremental removal didn't seem worth it)